crates/sui-metric-checker/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use anyhow::anyhow;
+use chrono::{DateTime, Duration, NaiveDateTime, Utc};
+use humantime::parse_duration;
+use serde::Deserialize;
+use strum_macros::Display;
+
+pub mod query;
+
+#[derive(Debug, Display, Deserialize)]
+pub enum QueryType {
+    Instant,
+    Range,
+}
+
+#[derive(Debug, Display, Deserialize)]
+pub enum ErrorCondition {
+    Greater,
+    Equal,
+    Less,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct QueryResultValidation {
+    // Threshold to report error on
+    pub threshold: f64,
+    // Program will report error if threshold violates condition specifed by this
+    // field.
+    pub condition: ErrorCondition,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Query {
+    // PromQL query to exeute
+    pub query: String,
+    // Type of query to execute
+    #[serde(rename = "type")]
+    pub query_type: QueryType,
+    pub validate_result: Option<QueryResultValidation>,
+    // Both start & end accepts formats
+    //  - %Y-%m-%d %H:%M:%S (UTC)
+    //  - now
+    //  - relative time offset before now, i.e. 1h 30m 10s
+    pub start: Option<String>,
```

Comment:
It seems in config files, start and end always expect to exist.

Reply:
Is it easy to move type, start and end into an enum on Instant vs Range? Or add a comment on when start and end can be None.